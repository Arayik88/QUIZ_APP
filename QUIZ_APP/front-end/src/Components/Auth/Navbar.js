import React,{useContext, useEffect} from 'react';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import Drawer from '@material-ui/core/Drawer';
import InputBase from '@material-ui/core/InputBase';
import SearchIcon from '@material-ui/icons/Search';
import Button from '@material-ui/core/Button';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import User from '../User/User';
import Admin from '../Admin/Admin';
import App from '../Exam/App';
import '../../Css/styles.css';
import { makeStyles, fade } from '@material-ui/core/styles';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from "react-router-dom";
import Search_List from '../User/Search_List';
import {UserContext} from '../../Contexts/user';
import store from "../../redux/store";
import { Provider } from "react-redux";
import {IdContext} from '../../Contexts/idcontext';
const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
    display: 'none',
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25),
    },
    marginLeft: 0,
   
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing(1),
      width: 'auto',
    },
  },
  searchIcon: {
    width: theme.spacing(7),
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  inputRoot: {
    color: 'inherit',
  },
  inputInput: {
    padding: theme.spacing(1, 1, 1, 7),
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      width: 200,
      '&:focus': {
        width: 250,
      },
    },
  },
}));

function Navbar() {
  //states
  
  const user =JSON.parse(localStorage.getItem("UserContext"));
  const classes = useStyles();
  const [state, Setstate] = React.useState(false);
  const [search, Setsearchstate] = React.useState("");
  const ID=useContext(IdContext);
  const toggleDrawer = (e) => () => Setstate(e)
 
   let x= user.roles[0];
   let roles=[];
   x.substr(0,x.length-1).substr(1).split(',').forEach(x=>{roles.push(x.trim())});
 
 

  return (
    <Router>
      
      <div >
        <AppBar position="static" >
          <Toolbar>
            <div className="menu">
              <IconButton style={{ padding: "0 10px 0 0" }} onClick={toggleDrawer(true)} ge="start" color="inherit" aria-label="menu">

                <MenuIcon />

              </IconButton>
            <div className="spacerphone"></div>
              <div className={classes.search}>
                <div className={classes.searchIcon}>
                  <SearchIcon />
                </div>
                <InputBase
               
                onChange={(e) => Setsearchstate(e.target.value)}
                  placeholder="Search…"
                  classes={{
                    root: classes.inputRoot,
                    input: classes.inputInput,
                  }}
                  inputProps={{ 'aria-label': 'search' }}
                />
              </div>
              <Link  style={{ color: 'white', textDecoration: 'none' }}  to={`/search/${search}`}>    <Button  color="inherit" >
                Search
                </Button> </Link>

              <Drawer open={state} onClose={toggleDrawer(false)}>
                <div className="drawer">
                


                  <Link style={{ color: 'white', textDecoration: 'none' }} to="/">    <Button onClick={toggleDrawer(false)} fullWidth color="primary" >
                    User
                </Button> </Link>


               { roles.includes("ADMIN") && <Link style={{ color: 'white', textDecoration: 'none' }} to="/admin">    <Button onClick={toggleDrawer(false)} fullWidth color="primary" >
                    Admin
                </Button> </Link>}
                { ID && <Link style={{ color: 'white', textDecoration: 'none' }} to="/exams">    <Button color="inherit" >
                Exams
                </Button> </Link>   }  

                  


                  <Link style={{ color: 'white', textDecoration: 'none' }} to="/">  <Button onClick={toggleDrawer(false)} fullWidth color="primary" >
                    Logout
                </Button></Link>
                </div>
              </Drawer>
            </div>
            <div className="navbar-menu">
             
              <Link style={{ color: 'white', textDecoration: 'none' }} to="/">    <Button color="inherit" >
                User
                </Button> </Link>
                
            { roles.includes('ADMIN') &&  <Link style={{ color: 'white', textDecoration: 'none' }} to="/admin">    <Button color="inherit" >
                Admin
                </Button> </Link>}
              { ID && <Link style={{ color: 'white', textDecoration: 'none' }} to="/exams">    <Button color="inherit" >
                Exams
                </Button> </Link>   }  
              
                    <Link style={{ color: 'white', textDecoration: 'none' }} to="/">    <Button onClick={()=>{localStorage.clear();window.location.replace("/");}} color="inherit" >
                Logout
                </Button> </Link>
                    <div className="spacer"></div>
              <div className={classes.search}>
                <div className={classes.searchIcon}>
                  <SearchIcon />
                </div>
                <InputBase
                  onKeyDown={()=>{console.log(this)}}
                  onChange={(e) => Setsearchstate(e.target.value)}
                  placeholder="Search…"
                  classes={{
                    root: classes.inputRoot,
                    input: classes.inputInput,
                  }}
                  inputProps={{ 'aria-label': 'search' }}
                />
              </div>
              
              <Link style={{ color: 'white', textDecoration: 'none' }} to={`/search/${search}`}>    <Button  color="inherit" >
                Search
                </Button> </Link>
            </div>
          </Toolbar>
        </AppBar>
       
        <Switch>

          <Route  exact path="/">
            <User />
          </Route>
           
            <Route path="/admin">
            <Admin></Admin>
            </Route>
            <Route  exact path= {`/search/${search}`}>
            <Search_List input={search} />
          </Route>
          <Route exact path='/exams'>
           <Provider store={store}>
           <App/> 
           </Provider>
           </Route>
        </Switch>
      </div>
    </Router>
  );
}
export default Navbar;