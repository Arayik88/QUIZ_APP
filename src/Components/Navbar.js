import React from 'react';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import Drawer from '@material-ui/core/Drawer';
import InputBase from '@material-ui/core/InputBase';
import SearchIcon from '@material-ui/icons/Search';
import Button from '@material-ui/core/Button';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import User from './User';
import '../Css/styles.css';
import { makeStyles, fade } from '@material-ui/core/styles';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,

} from "react-router-dom";
import Search_List from './Search_List';




const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
    display: 'none',
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25),
    },
    marginLeft: 0,
   
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing(1),
      width: 'auto',
    },
  },
  searchIcon: {
    width: theme.spacing(7),
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  inputRoot: {
    color: 'inherit',
  },
  inputInput: {
    padding: theme.spacing(1, 1, 1, 7),
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      width: 200,
      '&:focus': {
        width: 250,
      },
    },
  },
}));

function Navbar() {
  //states
  
  const classes = useStyles();
  const [state, Setstate] = React.useState(false);
  const [search, Setsearchstate] = React.useState("");

  const toggleDrawer = (e) => event => Setstate(e)
  const onKeyDown=(e,message)=>
  {  
    if(e.key==="Enter")
    {
      localStorage.setItem("input",message)
      window.location.href = '/search';
  
    }
  }
  
  return (
    <Router>
      <div >
        <AppBar position="static" >
          <Toolbar>
            <div className="menu">
              <IconButton style={{ padding: "0 10px 0 0" }} onClick={toggleDrawer(true)} ge="start" color="inherit" aria-label="menu">

                <MenuIcon />

              </IconButton>
              <div className="spacerphone"> </div>
              <div className={classes.search}>
                <div className={classes.searchIcon}>
                  <SearchIcon />
                </div>
                <InputBase
                onKeyDown={(e) => onKeyDown(e, search)}
                onChange={(e) => Setsearchstate(e.target.value)}
                  placeholder="Search…"
                  classes={{
                    root: classes.inputRoot,
                    input: classes.inputInput,
                  }}
                  inputProps={{ 'aria-label': 'search' }}
                />
              </div>

              <Drawer open={state} onClose={toggleDrawer(false)}>
                <div className="drawer">
                  <Link style={{ color: 'white', textDecoration: 'none' }} to="/">  <Button onClick={toggleDrawer(false)} fullWidth color="primary" >
                    Main
                </Button></Link>


                  <Link style={{ color: 'white', textDecoration: 'none' }} to="/users">    <Button onClick={toggleDrawer(false)} fullWidth color="primary" >
                    User
                </Button> </Link>


                  <Link style={{ color: 'white', textDecoration: 'none' }} to="/">  <Button onClick={toggleDrawer(false)} fullWidth color="primary" >
                    Exams
                </Button></Link>


                  <Link style={{ color: 'white', textDecoration: 'none' }} to="/">  <Button onClick={toggleDrawer(false)} fullWidth color="primary" >
                    Contact
                </Button></Link>


                  <Link style={{ color: 'white', textDecoration: 'none' }} to="/">  <Button onClick={toggleDrawer(false)} fullWidth color="primary" >
                    Logout
                </Button></Link>
                </div>
              </Drawer>
            </div>
            <div className="navbar-menu">
              <Link style={{ color: 'white', textDecoration: 'none' }} to="/">  <Button color="inherit" >
                Main
                </Button></Link>
              <Link style={{ color: 'white', textDecoration: 'none' }} to="/users">    <Button color="inherit" >
                User
                </Button> </Link>

              <Button color="inherit" >
                Exams
                </Button>

              <Button color="inherit"  >
                Contact
                    </Button>
              <div className="spacer"></div>
              <div className={classes.search}>
                <div className={classes.searchIcon}>
                  <SearchIcon />
                </div>
                <InputBase
                  onKeyDown={(e) => onKeyDown(e, search)}
                  onChange={(e) => Setsearchstate(e.target.value)}
                  placeholder="Search…"
                  classes={{
                    root: classes.inputRoot,
                    input: classes.inputInput,
                  }}
                  inputProps={{ 'aria-label': 'search' }}
                />
              </div>
              <Button color="inherit" >
                Logout
                    </Button>
            </div>
          </Toolbar>
        </AppBar>
       
        <Switch>

          <Route path="/users">
            <User />
          </Route>
          <Route  exact path="/">
            Hello
            </Route>
            <Route path="/search">
            <Search_List />
          </Route>
        </Switch>
      </div>

    </Router>
  );
}
export default (Navbar);