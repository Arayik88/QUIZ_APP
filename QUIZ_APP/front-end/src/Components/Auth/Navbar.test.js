import React from 'react';
import Navbar from './Navbar';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    localStorage.setItem("UserContext",JSON.stringify({
        "email": "string",
        "firstName": "string",
        "id": 0,
        "imageId": 0,
        "lastName": "string",
        "phone": "string",
        "roles":['[USER]']
        
      }))
  const tree = renderer
    .create(<Navbar ></Navbar>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});