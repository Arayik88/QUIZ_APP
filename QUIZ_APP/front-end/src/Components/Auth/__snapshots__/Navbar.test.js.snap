// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        className="menu"
      >
        <button
          aria-label="menu"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          disabled={false}
          ge="start"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "padding": "0 10px 0 0",
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
        <div
          className="spacerphone"
        />
        <div
          className="makeStyles-search-4"
        >
          <div
            className="makeStyles-searchIcon-5"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            className="MuiInputBase-root makeStyles-inputRoot-6"
            onClick={[Function]}
          >
            <input
              aria-label="search"
              className="MuiInputBase-input makeStyles-inputInput-7"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Search…"
              type="text"
            />
          </div>
        </div>
        <a
          href="/search/"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
              
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Search
            </span>
          </button>
           
        </a>
      </div>
      <div
        className="navbar-menu"
      >
        <a
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
              
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              User
            </span>
          </button>
           
        </a>
        <a
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
              
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Logout
            </span>
          </button>
           
        </a>
        <div
          className="spacer"
        />
        <div
          className="makeStyles-search-4"
        >
          <div
            className="makeStyles-searchIcon-5"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            className="MuiInputBase-root makeStyles-inputRoot-6"
            onClick={[Function]}
          >
            <input
              aria-label="search"
              className="MuiInputBase-input makeStyles-inputInput-7"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Search…"
              type="text"
            />
          </div>
        </div>
        <a
          href="/search/"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
              
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Search
            </span>
          </button>
           
        </a>
      </div>
    </div>
  </header>
  <div
    className="main-container"
  >
    <div
      className="general-row"
    >
      <div
        className="image-container-row"
      >
        <img
          alt="userImage"
          className="image"
          src={null}
        />
        <div
          style={
            Object {
              "height": 0,
              "overflow": "hidden",
            }
          }
        >
          <input
            id="fileInput"
            name="fileInput"
            onChange={[Function]}
            type="file"
          />
        </div>
        <div>
            
          <button
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "height": 50,
                "width": 50,
              }
            }
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
               
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
                focusable="false"
                role="presentation"
                style={
                  Object {
                    "height": 30,
                    "width": 30,
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 4v2h14V4H5zm0 10h4v6h6v-6h4l-7-7-7 7z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        className="about-container-row"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
              />
            </svg>
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm0 10.99h7c-.53 4.12-3.28 7.79-7 8.94V12H5V6.3l7-3.11v8.8z"
              />
            </svg>
          </span>
        </button>
        <div
          style={
            Object {
              "paddingRight": "20px",
            }
          }
        >
          <div
            className="about-container-list-item"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <div
                className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-describedby="Name-helper-text"
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiFilledInput-input"
                  disabled={false}
                  id="Name"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
              </div>
              <p
                className="MuiFormHelperText-root MuiFormHelperText-contained"
                id="Name-helper-text"
              >
                Name
              </p>
            </div>
          </div>
          <div
            className="about-container-list-item"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <div
                className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-describedby="Surname-helper-text"
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiFilledInput-input"
                  disabled={false}
                  id="Surname"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
              </div>
              <p
                className="MuiFormHelperText-root MuiFormHelperText-contained"
                id="Surname-helper-text"
              >
                Surname
              </p>
            </div>
          </div>
          <div
            className="about-container-list-item"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <div
                className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-describedby="Email-helper-text"
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiFilledInput-input"
                  disabled={false}
                  id="Email"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
              </div>
              <p
                className="MuiFormHelperText-root MuiFormHelperText-contained"
                id="Email-helper-text"
              >
                Email
              </p>
            </div>
          </div>
          <div
            className="about-container-list-item"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <div
                className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-describedby="Phone-helper-text"
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiFilledInput-input"
                  disabled={false}
                  id="Phone"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
              </div>
              <p
                className="MuiFormHelperText-root MuiFormHelperText-contained"
                id="Phone-helper-text"
              >
                Phone
              </p>
            </div>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Change
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="activeExams-container"
    >
      <h1>
        Active Exams
      </h1>
      <div
        className="activeExamsList-container"
      />
      <h1>
        Passed Exams
      </h1>
      <div
        className="activeExamsList-container"
      />
    </div>
  </div>
</div>
`;
